fem/pfespace.cpp

```diff
@@ -2881,6 +2882,7 @@ void ConformingProlongationOperator::Mult(const Vector &x, Vector &y) const
 
    const int out_layout = 0; // 0 - output is ldofs array
    gc.BcastEnd(ydata, out_layout);
+   y.Push();
```

Comment:
I'm double checking my code, but it seems that if Y has data from a shallow copy. This push will trigger the following error:  void *mfem::cuMemcpyHtoD_v2(void *, const void *, size_t): Assertion `0 && "Error in cuMemcpyHtoD"' failed when a device has been enabled.

Reply:
`y.Push()` just means update device version of the pointer, so should always work. Maybe the issue you are seeing will be addressed by https://github.com/mfem/mfem/pull/787?

Reply:
Yup, that is user error on our part. We will need to re-organize our code a bit and register our data with the memory manager. Thanks.